engine/src/flutter/lib/ui/platform_dispatcher.dart

```diff
@@ -1506,6 +1506,88 @@ class PlatformDispatcher {
   // configurationId does not match any configuration.
   @Native<Double Function(Double, Int)>(symbol: 'PlatformConfigurationNativeApi::GetScaledFontSize')
   external static double _getScaledFontSize(double unscaledFontSize, int configurationId);
+
+  /// Color palette preferred by the operating system UI widgets.
+  ///
+  /// On platforms that do not support a UI color palette this value is null.
+  ///
+  /// As of the current release, system colors are supported on web only.
+  ///
+  /// This is typically used in conjunction with
+  /// [AccessibilityFeatures.highContrast]. In particular, on Windows, when
+  /// a user enables high-contrast mode, they may also pick specific colors that
+  /// should be used by application user interfaces.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/system-color
+  ///   * https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors
+  Map<String, SystemColor>? get systemColors => null;
+
+  /// Standard system color names, as defined by W3C CSS specification.
+  ///
+  /// System color names in Flutter are case-sensitive. This is to make sure
+  /// color names can be easily used as [Map] keys. This is in contrast to CSS,
+  /// where system color names are not case-sensitive. That is, specifying
+  /// `background-color: aCcEnTcOlOr` is equivalent to specifying
+  /// `background-color: AccentColor`. In Flutter,
+  /// `systemColorNames['aCcEnTcOlOr']` will fail to find the respective color.
+  /// Always use the exact name string defined in this list.
+  ///
+  /// See also:
+  ///
+  ///   * https://drafts.csswg.org/css-color/#css-system-colors
+  static const List<String> systemColorNames = <String>[
```

Comment:
I always have a weird feeling these types of things should be `Set` just...on principle. But I know it just generates slightly slower code w/ no benefit. 🤷

Reply:
I'm actually starting to reconsider this list. Since the names are known and defined in the specification, I wonder if we should just make it an "enum" of sorts (it can't be literally enum because we detect their data dynamically).

Reply:
> (it can't be literally enum because we detect their data dynamically).

Can you elaborate? I also prefer enum as the dart:ui API

Reply:
Enums need to be const, but `SystemColor.value` is computed by reading it dynamically from the browser. However, we can make it an almost-enum, similar to `SemanticsAction`.